okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/NoDelayBackoff.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting3.okhttp;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+
+final class NoDelayBackoff implements BackoffStrategy {
+
+    private final int maxNumRetries;
+    private final AtomicInteger numTimesRetried = new AtomicInteger(0);
```

Comment:
this class is threadsafe, but it's unclear what 'threadsafe' is in this context. It being threadsafe leads readers to make assumptions about how it's used, which I _think_ are wrong.It might be worth just using a normal int and not doing the atomic thing.

Reply:
Fair, done.